src/gui/application.cpp

```diff
@@ -613,5 +616,29 @@ void Application::showSettingsDialog()
     _gui->slotShowSettings();
 }
 
+void Application::openPlaceholder(const QString &filename)
+{
+    QLatin1String placeholderExt(".owncloud");
+    if (!filename.endsWith(placeholderExt)) {
+        qWarning(lcApplication) << "Can only handle file ending in .owncloud. Unable to open" << filename;
+        return;
+    }
+    QString normalName = filename.left(filename.size() - placeholderExt.size());
+    auto folder = FolderMan::instance()->folderForPath(filename);
+    if (!folder) {
+        qWarning(lcApplication) << "Can't find sync folder for" << filename;
+        // TODO: show a QMessageBox for errors
+        return;
+    }
+    QString relativePath = QDir::cleanPath(normalName).mid(folder->cleanPath().length() + 1);
+    folder->downloadPlaceholder(relativePath);
+    auto con = QSharedPointer<QMetaObject::Connection>::create();
+    *con = QObject::connect(folder, &Folder::syncFinished, [con, normalName] {
+        QObject::disconnect(*con);
```

Comment:
Very cute... but understandable.

Reply:
Yeah, this emulate a "singleshot" connection.

However this needs some work because the file might not be successfully downloaded on the first sync, so i would need to show some feedback in that case.